numpy/core/fromnumeric.py

```diff
@@ -445,6 +445,10 @@ def put(a, ind, v, mode='raise'):
     array([ 0,  1,  2,  3, -5])
 
     """
+    if not isinstance(a, np.ndarray):
```

Comment:
@shoyer : Ah, fair enough.  I do see lots of similar wrappings in this file, so I'll follow suit then.  By the way, might you happen to know what that ```test_mem_insert``` is all about?  Even with your suggestion, that test will fail, but I'm not sure what it is or what the purpose of that test was.  I'm tempted to remove it, but it was located in the ```test_regression.py``` file, so I'm hesitant to do so until I have more information.

@mhvk : Thanks for your input as well!